### PR TITLE
Add metadata to results

### PR DESCRIPTION
**Pull request by buthter**
- Added metadata struct that keeps track of priority, functions, and categories for a single test
- Load a map of test names to TestMetadata from test_metadata.csv
- Include these metadata fields in the test output
- Refactor some related test result functions

### CODE REVIEW

**Comment by erademacher** [comment]
Accidentally logged into github with the wrong credentials... Oh well.

**Review by pvanderknyff — APPROVED** [review]
LGTM, though I would still wait for a review from Luke or Logan to make sure nothing unexpected will break in TDVT's guts.

**Comment by erademacher** [comment]
@lukewrites Would you mind taking a quick look to make sure I'm not doing anything obviously dumb here?

File: tdvt/tdvt/tdvt_core.py
Comment (lukewrites) [inline tdvt/tdvt/tdvt_core.py:100]: This is a great change 👍

**Pull request merged**
